File: Math/9613_GCD 합.c
#define _CRT_SECURE_NO_WARNINGS
#include <stdio.h>
#include <stdlib.h>

int gcd(int a, int b) {
	if (a % b == 0)
		return b;
	return gcd(b, a % b);
}

int main(void) {
	int T;
	scanf("%d", &T);
	while (T--) {
		int n;
		scanf("%d", &n);
		int* arr = malloc(sizeof(int) * n);
		for (int i = 0; i < n; i++) {
			scanf("%d", &arr[i]);
		}
		long long tot = 0;
		for (int i = 0; i < n; i++) {
			for (int j = i + 1; j < n; j++) {
				tot += gcd(arr[i], arr[j]);
			}
		}
		printf("%lld\n", tot);
		free(arr);
	}
	
	return 0;
}

File: Math/9613_GCD 합.py
#입출력
from sys import stdin
#입력 
input = stdin.readline

#최대공약수
def gcd(a,b): 
    if a % b == 0:
        return b
    else:
        return gcd(b,a%b)

#길이 입력
n=int(input())

#반복문 시작
for _ in range(n):
    #베열 입력
    arr=list(map(int,input().split()))
    
    tot=0

    for i in range(1,len(arr)):
        for j in range(i+1,len(arr)):
            tot += gcd(arr[i],arr[j])

    #출력
    print(tot)